### PR TITLE
mediaplayer: flags for subtitle & native video

### PR DESCRIPTION
**Pull request by psivasubramanian**
Enabling/Disabling subtitle & native video option

### CODE REVIEW

**Comment by eocanha** [comment]
Can we have some way to keep the current enabled behaviour if TEXT_SINK or NATIVE_VIDEO is not defined? Something like "#if !defined(TEXT_SINK) || ENABLED(TEXT_SINK)".

I suggest this because I saw that you added another pull request on meta-metrological to show these new options, but those of us who use buildroot won't have them, and therefore will have text sink and native video disabled by default.

**Comment by kraj** [comment]
Can we add option to buildroot for this as well ?

**Comment by psivasubramanian** [comment]
@eocanha how can we add this to buildroot as Khem suggested?

File: Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp [hunk 1]
```diff
@@ -2158,10 +2158,21 @@ void MediaPlayerPrivateGStreamer::createGSTPlayBin()
...
+#if ENABLE(TEXT_SINK)
     unsigned flagText = getGstPlayFlag("text");
+#else
+    unsigned flagText = 0x0;
+#endif
+    
     unsigned flagAudio = getGstPlayFlag("audio");
     unsigned flagVideo = getGstPlayFlag("video");
+    
+#if ENABLE(NATIVE_VIDEO)
```
Comment (vertexodessa) [inline Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp]: we should define these options to make them available for the build 
https://github.com/Metrological/WebKitForWayland/blob/master/Source/cmake/OptionsWPE.cmake

**Comment by psivasubramanian** [comment]
Thanks @vertexodessa . Added this option to buildroot with default enabled.

**Comment by eocanha** [comment]
After merging this pull request, I wasn't sure if having the option to manage these new options from buildroot menuconfig was interesting or not. I've created https://github.com/Metrological/buildroot-wpe/pull/26 . If it's interesting, we can just merge on buildroot.

**Pull request merged**